### PR TITLE
KAFKA-9992: EmbeddedKafkaCluster.deleteTopicAndWait not working with kafka_2.13

### PR DESCRIPTION
**Pull request by akatona84**
Eliminating JavaConverters in EmbeddedKafkaCluster

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by akatona84** [comment]
@vvcephei without the scala to java converter it would be compatible.

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java [hunk 1]
```diff
@@ -314,8 +321,7 @@ private TopicsDeletedCondition(final Collection<String> topics) {
 
         @Override
         public boolean conditionMet() {
-            final Set<String> allTopics = new HashSet<>(JavaConverters.setAsJavaSetConverter(
-                brokers[0].kafkaServer().zkClient().getAllTopicsInCluster(false)).asJava());
+            final Set<String> allTopics = new HashSet<>(getAllTopics());
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java]: It looks like we don’t need to make a copy here anymore

**Review by vvcephei — APPROVED** [review]
Hey, thanks for this, @akatona84 !

Just one minor comment, otherwise it lgtm (pending tests)

**Comment by vvcephei** [comment]
Test this please

**Comment by vvcephei** [comment]
(That last comment was to trigger the tests)

**Review by abbccdda — APPROVED** [review]
LGTM as well!

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java [hunk 2]
```diff
@@ -297,6 +295,15 @@ public void deleteAllTopicsAndWait(final long timeoutMs) throws InterruptedExcep
         }
     }
 
+    public Set<String> getAllTopics() {
+        scala.collection.Iterator<String> topicsIterator = brokers[0].kafkaServer().zkClient().getAllTopicsInCluster().iterator();
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java]: > getAllTopicsInCluster()

```getAllTopicsInCluster(false)```

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java [hunk 3]
```diff
@@ -297,6 +295,15 @@ public void deleteAllTopicsAndWait(final long timeoutMs) throws InterruptedExcep
         }
     }
 
+    public Set<String> getAllTopics() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java]: ```getAllTopicsInCluster``` seems to be duplicate to this new method. Could we remove the former?

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java (hunk 2 above)
Comment (akatona84) [reply streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java]: thx, i was cherry picking my change from 2.4.1 based kafka branch and config resolution did not go well, I uploaded a new, clean change.

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java (hunk 1 above)
Comment (akatona84) [reply streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java]: fixed in new commit

**Comment by akatona84** [comment]
retest this please

**Comment by vvcephei** [comment]
test this please

**Comment by vvcephei** [comment]
test this please

**Review by vvcephei — APPROVED** [review]
LGTM. Thanks for the contribution, @akatona84 !

I'll just wait for the tests to pass.

**Review by chia7712 — APPROVED** [review]

**Comment by vvcephei** [comment]
The tests that failed are known to be flaky and seem unrelated to this change:
    org.apache.kafka.streams.integration.EosBetaUpgradeIntegrationTest.shouldUpgradeFromEosAlphaToEosBeta[true]
    org.apache.kafka.streams.integration.EosBetaUpgradeIntegrationTest.shouldUpgradeFromEosAlphaToEosBeta[true]
    org.apache.kafka.streams.integration.EosIntegrationTest.shouldNotViolateEosIfOneTaskFailsWithState[exactly_once]

**Pull request merged**